src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java

```diff
@@ -0,0 +1,185 @@
+
+        while(it.hasNext()) {
+            OntClass oc = (OntClass)it.next();
+            if (!oc.isAnon()) {
+                setResources(oc.getURI(), oc.getLocalName());
+                this.classesToGenerate.add(oc.getURI());
+            }
+        }
+    }
+
+
+    /**
+     * Creates the ontology model based on the ontology argument
+     * @param processImports
+     * @return
+     */
+    private OntModel createOntModel(boolean processImports) {
+        OntModelSpec s = new OntModelSpec(OntModelSpec.OWL_MEM);
+
+        OntModel ontModel = ModelFactory.createOntologyModel(s,null);
+        OntDocumentManager dm = ontModel.getDocumentManager();
+        dm.setProcessImports(processImports);
+        return ontModel;
+    }
+
+    private void setResources(String uri, String resourceName) {
+        OWGClass owgClass = new OWGClass(uri, resourceName);
+        owgClassList.add(owgClass);
+
+    }
+
+    /**
+     * Will create java classes with resources from a given set of ontologies.
+     * @param args
+     * @throws IOException
+     */
+    public static void main(String[] args) throws IOException {
+
+        for (String arg : args) {
+
+            OntologyGeneration ctx = new OntologyGeneration();
+
+            ctx.addOntologyToGenerate(new FileInputStream(arg));
+
+            owgMapperWriter();
+
+            owgClassList.clear();
+        }
+    }
+
+    /**
+     * Writes the generated Java classes
+     * @throws IOException
+     */
+    private static void owgMapperWriter() throws IOException {
+
+        OWGClass test = owgClassList.get(0);
+        String classNameTest = test.uri;
+        String variableClassName = classNameTest.substring(classNameTest.lastIndexOf('/') + 1, classNameTest.indexOf('#'));
+
+     //   String variableClassName = "hello";
+        List<String> lines = owgMapperInfo(variableClassName);
+        Path file = Paths.get("src/main/java/com/ncc/aif/" + variableClassName + ".java");
+        Files.write(file, lines, Charset.forName("UTF-8"));
+    }
+
+    private static List<String> owgMapperInfo(String variableClassName) {
+
+        List<String> owgMapping = new ArrayList<>();
+
+        owgMapping.addAll(owgMapperHeader());
+
+        String className = "public final class " + variableClassName + " {";
+        owgMapping.add(className);
+
+        for (OWGClass owgClass : owgClassList) {
+
+            String name = owgClass.getName();
+
+            if (name.contains(".") || name.contains("-")) {
+                name = name.replace(".", "_");
+                name = name.replace("-", "_");
+            }
+
+            String resource = "    public static final Resource " + name + " = ResourceFactory.createResource(\"" + owgClass.getUri() + "\");";
+            owgMapping.add(resource);
+        }
+
+        String ending = "}";
+        owgMapping.add(ending);
+
+        return owgMapping;
+    }
+
+    private static List<String> owgMapperHeader () {
+
```

Comment:
I think it would be nice to have a brief, generated file and/or class comment that:
- Mentions that this is a generated file (and "DO NOT EDIT" warning);
- Includes the date of generation;
- Gives a one-liner on what the file is and/or refers to the README.
- If it's available programmatically, the ontology’s comment, label, version info (rdfs:comment, rdfs:label, owl:versionInfo).

Reply:
For the record:  the comment, label, and version info weren't easily obtainable programmatically.  Resolving this comment.